### PR TITLE
Archaius 2.x

### PR DESCRIPTION
**Pull request by elandau**
WORK IN PROGRESS

This is the initial implementation of Archaius v2 without a dependency on apache configuration. This implementation is far simpler, has no static state and uses Typesafe Config for loading of external resources.

This code is not yet at feature parity with 1.0 but is at a state where we can start discussing the merits of this approach. In addition we will be creating an archaius-legacy subproject with bindings to the older static API.

### CODE REVIEW

File: archaius-core/src/main/java/netflix/archaius/CascadeStrategy.java [hunk 1]
```diff
@@ -0,0 +1,29 @@
+package netflix.archaius;
```
Comment (spencergibb) [inline archaius-core/src/main/java/netflix/archaius/CascadeStrategy.java]: Curious why the new package structure?  Why drop the `com.`?

**Comment by spencergibb** [comment]
:+1: looking forward to this.

File: archaius-core/src/main/java/netflix/archaius/CascadeStrategy.java (hunk 1 above)
Comment (elandau) [inline archaius-core/src/main/java/netflix/archaius/CascadeStrategy.java]: There's a growing trend to remove the com. as it is completely unnecessary.  :)

Comment (gorzell) [inline archaius-core/src/main/java/netflix/archaius/CascadeStrategy.java]: It's only completely unnecessary if you have a monopoly on all of the possible domains. ;-)

**Pull request merged**